packages/nuxt/src/app/nuxt.ts

```diff
@@ -254,9 +254,9 @@ export function createNuxtApp (options: CreateOptions) {
     static: {
       data: {},
     },
-    runWithContext (fn: any) {
-      if (nuxtApp._scope.active) {
-        return nuxtApp._scope.run(() => callWithNuxt(nuxtApp, fn))
+    runWithContext (fn: any, scope = nuxtApp._scope) {
+      if (scope.active && !getCurrentScope()) {
```

Comment:
I think we should probably try to run with `getCurrentInstance() || getCurrentScope()` to ensure it's scoped as locally as possible. Then I think we can leave `defineNuxtComponent` unchanged.

This might also be the source of https://github.com/nuxt/nuxt/issues/26820, and (if so) we can probably revert the change there.

Reply:
Do you mean `if (scope.active && !(getCurrentInstance() || getCurrentScope())) {`?

Reply:
And do you think we can complete remove second param in such case?

Reply:
_Ideally_.

Reply:
This was truly the cause! I'll roll it back then.